src/main/kotlin/uk/gov/dluhc/printapi/service/BankHolidaysDataService.kt

```diff
@@ -0,0 +1,35 @@
+package uk.gov.dluhc.printapi.service
+
+import mu.KotlinLogging
+import org.springframework.stereotype.Service
+import uk.gov.dluhc.bankholidaysdataclient.BankHolidayDataClient
+import uk.gov.dluhc.bankholidaysdataclient.BankHolidayDivision
+import java.time.LocalDate
+
+private val logger = KotlinLogging.logger {}
+
+/**
+ * Service class that uses internal dluhc library client which interacts with S3 bank holidays bucket.
+ * Uses v2 version of AWS S3 BankHolidayDataClient.
+ * Note: The service method that calculates bank holidays is cached.
+ */
+@Service
+class BankHolidaysDataService(
+    private val bankHolidayDataClient: BankHolidayDataClient,
+) {
+
+    fun getUpcomingBankHolidays(
+        gssCode: String,
+        fromDate: LocalDate,
+        toDate: LocalDate = fromDate.plusDays(100)
+    ): List<LocalDate> {
```

Comment:
To align this function signature with vca-api function signature, I will raise another `vca-api` PR.